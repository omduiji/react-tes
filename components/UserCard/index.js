import React, { useContext, useEffect, useState } from 'react';
import { usersContext } from '../../stores/users';
import { getDisplayName } from '../../utils';
import Button from '../Button';
import Input from '../Input';
import styles from './styles.module.scss';

const UserCard = ({ user }) => {
  const [editUser, setEditUser] = useState(user);
  const [isEditMode, setIsEditMode] = useState(false);
  const [, dispatch] = useContext(usersContext);
  const handleClick = (e) => {
    isEditMode ? setIsEditMode(false) : setIsEditMode(true);
  };
  useEffect(() => {
    setEditUser(user);
  }, [user]);

  if (isEditMode) {
    return (
      <div className={styles.card}>
        <Input
          onChange={(value) => setEditUser({ ...editUser, first_name: value })}
          label="First name"
          value={editUser.first_name}
        />
        <Input
          onChange={(value) => setEditUser({ ...editUser, last_name: value })}
          label="Last name"
          value={editUser.last_name}
        />
        <Input
          onChange={(value) => setEditUser({ ...editUser, title: value })}
          label="Title"
          value={editUser.title}
        />
        <Input
          onChange={(value) => setEditUser({ ...editUser, company: value })}
          label="Company"
          value={editUser.company}
        />
        <Button handleClick={handleClick}>Save</Button>
      </div>
    );
  }

  return (
    <div className={styles.card}>
      <img src={editUser.photo} />
      <h1>{getDisplayName(editUser)}</h1>
      {user.title && <p className={styles.title}>{editUser.title}</p>}
      {user.company && <p className={styles.company}>{editUser.company}</p>}
      <Button handleClick={handleClick}>Edit</Button>
    </div>
  );
};

export default UserCard;
