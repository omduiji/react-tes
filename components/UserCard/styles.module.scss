.card {
    max-width: 100vw;
    order: 1;
    position: sticky;
    top: 0;
    background-color: white;
    padding: 20px;
    width: 100%;
    border-bottom: 1px solid #eee;

    @media (min-width: 500px) {
        max-width: 68vw;
        order: 2;
        position: static;
        border-left: 1px solid #eee;
        border-bottom: none;
    }

    p {
        margin: 0;
        font-size: 1.6rem;
    }

    img {
        width: 200px;
        height: 200px;
    }
}
