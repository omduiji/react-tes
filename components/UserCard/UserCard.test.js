import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom'
import UserCard from '.'

const testUser = {
    id: 42,
    photo: 'test.jpg',
    first_name: 'Jane',
    last_name: 'Doe',
}

test('should render UserCard', () => {
  const { queryByText, asFragment } = render(<UserCard user={testUser} />)

  expect(queryByText('JaneDoe')).toBeInTheDocument()

  expect(asFragment()).toMatchSnapshot()
})

test.skip('should activate edit mode', () => {
    const { queryByText, asFragment, getByRole } = render(<UserCard user={testUser} />)

    fireEvent.click(queryByText('Edit'))

    expect(queryByText('Edit')).not.toBeInTheDocument()
    expect(queryByText('Save')).toBeInTheDocument()

    expect(asFragment()).toMatchSnapshot()
})
