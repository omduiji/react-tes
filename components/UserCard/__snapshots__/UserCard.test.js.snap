// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserCard 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <img
      src="test.jpg"
    />
    <h1>
      JaneDoe
    </h1>
    <button
      type="button"
    >
      Edit
    </button>
  </div>
</DocumentFragment>
`;
